software/main.py

```diff
@@ -0,0 +1,266 @@
+                    
+                except:
+                    print("Ei leidnud")
+                    
+                
+            
+            if frame%30 == 0:
+                frame = 0
+                end = time.time()
+                fps = 30/(end-start)
+                start = end
+                #[Object: x=7; y=212; size=15.0; distance=212; exists=True]  
+                print(fps)  
+
+            if state == "find_ball":
+                print("--------Searching ball--------")
+                if len(processedData.balls)!=0:
+                    state="move"
+                else:
+                    if ball_side==1:
+                        robot.find_ball(spin)
+                    elif ball_side==0:
+                        robot.find_ball((-1)*spin)
+                        
+            elif state == "move":
+                print("--------Moving to ball--------")
+                if (len(processedData.balls)!=0):
+                    
+                    processedData.balls.sort(key= lambda x: x.size)
+                    targeted_ball=processedData.balls[-1]
+                    xcord=targeted_ball.x
+                    ycord=targeted_ball.y
+                    dist=targeted_ball.distance
+                    delta=((xcord-reso_x_mid)/reso_x_mid)
+                    print(dist)
+                        #Follows where is ball so find_ball function is more efficient
+                    if delta<0:
+                        ball_side = 0
+                    else:
+                        ball_side = 1
+                    
+                        #SpeedY based on ball distance
+                    if dist > 175:
+                        speedY=0.2
+                    else:
+                        speedY=0.5
+                        #Controlls that balls location is ready for robot's orbit function
+                    if xcord < (reso_x_mid + 15) and xcord >(reso_x_mid -15) and dist > 250:
+                        #robot.center_ball(xcord)
+                        state="orbit"
+                    else:
+                        #Moves to ball
+                        speedR = delta*(-1.75)
+                        speedX = delta*0.25
+                        robot.move(speedX, speedR, speedY)
+                else:
+                    #If there is no ball
+                    state="find_ball"
+                        
+            elif state=="orbit":
+                print("--------Centering ball and basket--------")
+                
+                
+                #If there is ball
+                if len(processedData.balls)>0:
+                    processedData.balls.sort(key= lambda x: x.size)
+                    targeted_ball=processedData.balls[-1]
+                    xcord=targeted_ball.x
+                    ycord=targeted_ball.y
+                    dist=targeted_ball.distance
+                    delta=((xcord-reso_x_mid)/reso_x_mid)
+
+                    if (xcord > 434):

```

Comment:
I suggest using multipliers of the camera resolution. Additionally the variable rotate_dir is named in a confusing way. Does it define the roation direction or rotation speed?